app/graphql/types/publish_target_type.rb

```diff
@@ -4,11 +4,11 @@ module Types
   class PublishTargetType < Types::BaseObject
     description 'A publish target'
 
-    field :string_key, ID, null: false
-    field :display_label, String, null: false
+    field :type, Enums::PublishTargetTypeEnum, null: false, method: :target_type
     field :publish_url, String, null: false
     field :api_key, String, null: false
     field :doi_priority, Integer, null: false
     field :is_allowed_doi_target, Boolean, null: false
+    field :string_identifier, String, null: false, method: :string_identifier
```

Comment:
stringIdentifier is project string key and target type joined together with an underscore. Example: academic_commons_production

Reply:
Building off of Ben's question, `string_key` has widespread use throughout the app and has a fairly well understood meaning, but `string_identifier` may be confusing because it sounds similar and string keys are used as identifiers in certain contexts.  How would we all feel about renaming this to something like `combined_key` or `combined_identifier`, to indicate the combined type and project key origin of the value?

Reply:
It might be good to just ticket discussing this later - I _think_ that if the the `string_identifier` always ends in the same value as `target_type`, that might signal something about whether we need to expose separately in the gql API, or maybe just expose `target_type` and `project_string_key`?

Reply:
This `string_identifier` is used in CSV data and the Digital Object update API for `publish_entries` hash keys, but requiring users to specify separate project string key and target_type values in two columns in a CSV context (or two fields in a JSON api context) is exactly why we went for the combined key concept -- to simplify addition and removal of publish targets and also to maintain the format that people are used to.

If we're sticking with that decision, I just feel like it's easier to make a field name change while all occurrences of this new field are clearly outlined in the PR, rather than merging the code in and having to dig around for where else it may be used in the future.

Reply:
but this is the GraphQL api? The string_identifier concept makes sense pragmatically for CSVs, but every where else I feel like we would want to decompose it (tho clearly that is not a universal feeling).

Reply:
I'm glad to talk more about whether we want to remove `string_identifier` from the GraphQL API, but that feels separate from the point that I'm trying to make.  I just mean that if this concept of `string_identifier` continues to exist in the app (e.g. as `string_identifier` and `parse_string_identifier` methods in the PublishTarget class), I'm proposing that we go with a different term.

Reply:
@cgalarza I just spoke with Ben about this and since tomorrow's your last day, I can take over this PR and make the field name change (and other proposed changes, if @barmintor is okay with that).

Reply:
@elohanlon That makes sense!